### PR TITLE
Cron and time / day functions ES2015 module

### PR DESCRIPTION
**Pull request by Alys**
**Incomplete**. @crookedneighbor It's an early PR in case you want to pick up anything I'm doing wrong already.

It refactors `dayMapping`, `sanitizeOptions`, `startOfWeek`, and `startOfDay` from `common/script/index.js` to a new module `common/script/cron.js`.

Still to come: `daysSince`, `shouldDo`, and `cron`, and any others that I find that are related.
## lint errors - help requested

I had to comment out three of the lines from `.eslintrc` before the tests would run.

`"no-empty-pattern": 2,` and `"no-arrow-condition": 2,` were giving these errors:

```
  1:1  error  Definition for rule 'no-empty-pattern' was not found    no-empty-pattern
  1:1  error  Definition for rule 'no-arrow-condition' was not found  no-arrow-condition
```

and `"func-style": [2, "declaration", { "allowArrowFunctions": true }],` was giving this:

```
Error: /vagrant/.eslintrc:
    Configuration for rule "func-style" is invalid:
    Value "2,declaration,[object Object]" has more items than allowed.
```

Is there something I need to do locally to get them to work? I did a rebase and an `npm install` yesterday so I should have the latest of everything.

### CODE REVIEW

**Comment by crookedneighbor** [comment]
Hmmm... there must be something funky about eslint and vagrant.

File: common/script/cron.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
+/*
+  ------------------------------------------------------
+  Cron and time / day functions
+  ------------------------------------------------------
+ */
+import _ from 'lodash';
+
+let moment = require('moment');
```
Comment (paglias) [inline common/script/cron.js]: import?

File: common/script/cron.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+  // return a new object, we don't want to add "now" to user object
+  return {
+    dayStart,
+    timezoneOffset,
+    now,
+  };
+}
+
+export function startOfWeek (options) {
```
Comment (paglias) [inline common/script/cron.js]: you can use a default here: startOfWeek (options = {}) {

File: common/script/cron.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+
+/*
+  This is designed for use with any date that has an important time portion (e.g., when comparing the current date-time with the previous cron's date-time for determing if cron should run now).
+  It changes the time portion of the date-time to be the Custom Day Start hour, so that the date-time is now the user's correct start of day.
+  It SUBTRACTS a day if the date-time's original hour is before CDS (e.g., if your CDS is 5am and it's currently 4am, it's still the previous day).
+  This is NOT suitable for manipulating any dates that are displayed to the user as a date with no time portion, such as a Daily's Start Dates (e.g., a Start Date of today shows only the date, so it should be considered to be today even if the hidden time portion is before CDS).
+ */
+
+export function startOfDay (options) {
```
Comment (paglias) [inline common/script/cron.js]: same here for the default value

**Comment by Alys** [comment]
Thanks paglias, I've made those changes.

I've added daysSince.

**Comment by Alys** [comment]
Oh heck no. I thought we weren't merging these to develop yet. I'm not confident in the preen changes without tests so I'm going to revert them.

**Comment by Alys** [comment]
The two most recent commits from this PR (for preenHistory refactoring) have been reverted in https://github.com/HabitRPG/habitrpg/commit/69553260d322c68975eddcc25161c05ec392c41b

**Pull request merged**